man/rep-xgettext.1
.\" This manpage has been automatically generated by docbook2man 
.\" from a DocBook document.  This tool can be found at:
.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
.\" Please send any bug reports, improvements, comments, patches, 
.\" etc. to Steve Cheng <steve@ggi-project.org>.
.TH "REP-XGETTEXT" "1" "04 avril 2003" "" ""

.SH NAME
rep-xgettext \- extract i18n strings from lisp scripts.
.SH SYNOPSIS

\fBrep-xgettext\fR [ \fB--include \fIDEFINER\fB\fR ] [ \fB--c \fR ] [ \fB--pot \fR ]

.SH "DESCRIPTION"
.PP
\fBrep-xgettext\fR The usual method of constructing
message catalogue templates (.pot files) is to run xgettext on the C
source files of the program (that have been annotated for i18n). 
librep provides the rep-xgettext program to perform the same task for
files of Lisp code.
.SH "OPTIONS"
.TP
\fB--include\fR
include DEFINER. This is not documented by upstream.
.TP
\fB--c\fR
This makes it output pseudo C code containing the string constants found.
.TP
\fB--pot\fR
This makes it output in standart .pot file.
.SH "SEE ALSO"
.PP
The programs are documented fully by \fIJohn
Harper\fR available via the \fBInfo\fR system.
.SH "AUTHOR"
.PP
This manual page was written by Christian Marillat <marillat@debian.org> for
the Debian GNU/Linux system (but may be used by others).


ChangeLog
2009-10-24  Christopher Bratusek <zanghar@freenet.de>
	* librep.spec.in
	* man/Makefile.in
	* man/rep.1
	* man/repdoc.1
	* man/rep-remote.1
	* man/rep-xgettext.1: added man-pages from Debian

	* debian/patches/00list
	* debian/changelog.in
	* debian/control
	* debian/compat
	* debian/copyright
	* debian/librep9.install.standard
	* debian/librep9.preinst
	* debian/librep9.symbols
	* debian/librep-dbg.links
	* debian/librep-dev.install.standard
	* debian/librep-dev.links
	* debian/librep-dev.lintian-overrides 
	* debian/librep-dev.preinst
	* debian/README.source
	* debian/rep-doc.info
	* debian/rep-doc.install
	* debian/rep.install
	* debian/rep.links
	* debian/rules
	* debian/watch: added debian packaging scripts based on official one

2009-10-10  Christopher Bratusek <zanghar@freenet.de>
	* src/files.c
	* src/repint.h
	* src/repint_subrs.h
	* src/rep_subrs.h: added file-executable-p function

	* man/news.texi
	* man/lang.texi: updated

2009-09-24  Christopher Bratusek <zanghar@freenet.de>
	* librep.ebuild.in: improved [Fuchur] [Christopher Bratusek]

2009-09-13  Christopher Bratusek <zanghar@freenet.de>
	* librep.doap: added

2009-09-12  Christopher Bratusek <zanghar@freenet.de>
	* configure.in: bump version to 0.90.3
	                s/can't/cannot/
	                s/don't/do not/
			to prevent syntax highlighting errors (eg: in geany/vim)
			small improvements in a test [FreeBSD]

	* src/Makefile.in
	* src/rep-config.sh: removed rep-config script (use librep.pc instead)

2009-09-10  Christopher Bratusek <zanghar@freenet.de>
	* configure.in: fallback to old ffi check if there's no libffi.pc [Vincent Untz]

	* man/news.texi: updated

2009-09-05  Christopher Bratusek <zanghar@freenet.de>
	* librep.spec.in: improved [Kim B. Heino]

2009-08-25  Christopher Bratusek <zanghar@freenet.de>
	* configure.in: post-release version bump to 0.90.2

	* src/lisp.c: fixed an major defunct in prin1 + utf8 [Timo Korvola]

	* src/streams.c: wrong description of formats %s and %S

2009-08-22  Christopher Bratusek <zanghar@freenet.de>
	* src/rep.c: properly terminate rep interpreter [Jürgen Hötzel]

	* src/readline.c: save/load readline history in interactive mode [Jürgen Hötzel]

	* src/lisp.c
	* src/uft8.c: No C++ style comments in C code

2009-07-25  Christopher Bratusek <zanghar@freenet.de>
	* librep.pc.in: add -L$prefix to libs

2009-07-24  Christopher Bratusek <zanghar@freenet.de>
	* emulate-gnu-tar
	* lisp/rep/io/file-handlers/tar.jl: added support for tar.xz and tar.lzma

	* lisp/rep/io/file-handlers/tar.jl: improved regex for parsing tar output [Alexey I. Froloff]

	* src/ffi.c: improved ffi-binding [Alexey I. Froloff]

	* src/librep.sym: added some missing entries [Alexey I. Froloff]

2009-07-15  Christopher Bratusek <zanghar@freenet.de>
	* configure.in
	* Makedefs.in
	* src/Makefile.in: update checks for libffi

2009-07-11  Christopher Bratusek <zanghar@freenet.de>
	* src/unix_processes.c: fix a definition

2009-07-02  Christopher Bratusek <zanghar@freenet.de>
	* configure.in: obligatory post-release version bump

	* .git: created 0.90.0 tag and pushed 0.17.4 tag

2009-06-30  Christopher Bratusek <zanghar@freenet.de>
	* man/lang.texi: updated for rep.util.utf8 [Teika Kazura]

	* src/lispcmds.c: fixed a typo [Teika Kazura]

	* src/utf8.c: improved copyright notice, removed unused macros [Teika Kazura]

2009-06-19  Christopher Bratusek <zanghar@freenet.de>
	* src/unix_processes
	* src/sdbm.c
	* src/sdbm_pair.c: removed 3 unused variables

2009-05-30  Christopher Bratusek <zanghar@freenet.de>
	* src/utf8.c: added UTF-8 Support! [Wang Diancheng]

2009-05-06  Christopher Bratusek <zanghar@freenet.de>
	* Makefile.in: remove tar target [Ritz]

	* configure.in: 0.17.4 released, bump to 0.90.0

	* lisp/scheme/
	* lisp/scheme.jl
	* lisp/unscheme/
	* lisp/unscheme.jl
	* lisp/rep/vm/compiler/scheme.jl
	* lisp/rep/vm/compiler/unscheme.jl: removed

	* lisp/rep/vm/compiler.jl
	* lisp/rep/user.jl
	* lisp/Makefile: updated

2009-05-05  Christopher Bratusek <zanghar@freenet.de>
	* man/lang.tex
	* man/librep.texi
	* man/news.texi: 'Numbers' section is revised, mainly on machine dependence. Minor improvements [Teika]

	* configure.in: small configures ending message update (from rep-gtk/sawfish)

2009-03-24  Christopher Bratusek <zangahr@freenet.de>
	* Makedefs.in: first define datarootdir then stuff that refers to it

	* man/news.texi: updated

2009-03-06  Christopher Bratusek <zanghar@freenet.de>
	* configure.in
	* Makedefs.in
	* intl/Makefile.in: don't ignore datarootdir setting

2009-03-05  Christopher Bratusek <zanghar@freenet.de>
	* configure.in: 0.17.3 released, bump to 0.17.4

2009-02-18  Christopher Bratusek <zanghar@freenet.de>
	* lisp/rep/io/file-handlers/tar.jl: BSD-Tar is not supported by librep, give users a usefull warning message [Mark Diekhans]

2009-02-03  Christopher Bratusek <zanghar@freenet.de>
	* install-sh: added for libtool < 2.2

	* librep.spec.in: don't try to package librep.a and %{_infodir}/dir

	* autogen.sh: re-enable the use of autoheader

	* configure.in: fix templates (for autoheader)

2009-02-02  Christopher Bratusek <zanghar@freenet.de>
	* librep.pc.in: remove bogus entries

	* autogen.sh: don't use bash specific syntax [Mark Diekhans]

	* configure.in: improved ending message

2009-02-01  Christopher Bratusek <zanghar@freenet.de>
	* MAINTAINERS: updated my mail address

	* autogen.sh: work around an issue

	* librep.spec.in: - don't try to package non-existent rep.m4
			  - make sure emacs_lispdir is defined

2009-01-18  Christopher Bratusek <zanghar@freenet.de>
	* librep.spec.in: - update BuildRequires 
	                  - merge changes from ritz's spec-file

2009-01-02  Christopher Bratusek <nano-master@gmx.de>
	* librep.spec.in: source-archive is .tar.bz2

2009-01-01  Christopher Bratusek <nano-master@gmx.de>
	* autogen.sh: reworked

	* m4: add m4 directory to shut up libtool 1.x

	* configure.in: begin rework

	* install-sh: don't ship this file


src/rep-xgettext.jl
#! /usr/bin/rep
!#

;; rep-xgettext.jl -- extract i18n strings from lisp scripts
;; $Id: rep-xgettext.jl,v 1.8 2000/08/03 16:17:09 john Exp $

;; This file is part of librep.

;; librep is free software; you can redistribute it and/or modify it
;; under the terms of the GNU General Public License as published by
;; the Free Software Foundation; either version 2, or (at your option)
;; any later version.

;; librep is distributed in the hope that it will be useful, but
;; WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
;; GNU General Public License for more details.

;; You should have received a copy of the GNU General Public License
;; along with librep; see the file COPYING.  If not, write to
;; the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.

(require 'rep.i18n.xgettext)

(defvar *write-c-file* nil)


;; entry point

(when (get-command-line-option "--help")
  (write standard-output "\
usage: rep-xgettext [OPTIONS...] FILES...

where OPTIONS are any of:

  --include DEFINER
  --c
  --pot\n")
  (throw 'quit 0))

(when (or (get-command-line-option "-c") (get-command-line-option "--c"))
  (setq *write-c-file* t))
(when (or (get-command-line-option "-p") (get-command-line-option "--pot"))
  (setq *write-c-file* nil))

(let ((included '()) tem)
  (while (setq tem (get-command-line-option "--include" t))
    (setq included (cons (intern tem) included)))
  (when included
    (set-included-definers included)))

(while command-line-args
  (let ((file (car command-line-args)))
    (setq command-line-args (cdr command-line-args))
    (scan-file file)))

(if *write-c-file*
    (output-c-file)
  (output-pot-file))

;; Local variables:
;; major-mode: lisp-mode
;; End:
